### PR TITLE
Refactor api code

### PR DESCRIPTION
**Pull request by AyaAljalees**

### CODE REVIEW

**Review by Amal-Mousa — APPROVED** [review]

**Review by KhaledToha — APPROVED** [review]

**Review by Mohammed-Sbeata — APPROVED** [review]
Good Job 👏

File: src/controllers/auth.ts [hunk 1]
```diff
@@ -36,13 +40,7 @@ const signup = (req: Request, res: Response, next: NextFunction): void => {
...
       data: [{ name, email, phone }],
     }))
-    .catch((err: CustomError | joiInterface) => {
-      if ('isJoi' in err) {
-        next(new CustomError(406, err.details[0].message));
-      } else {
-        next(new CustomError(500, 'server error'));
-      }
-    });
+    .catch((error)=>next(error));
```
Comment (KhaledToha) [inline src/controllers/auth.ts:43]: we should determine if the error came from validation or else
so make it like this 
.catch((err: CustomError | joiInterface) => {
      if ('isJoi' in err) {
        next(new CustomError(406, err.details[0].message));
      } else {
        next(err);
      }
    });

File: src/controllers/auth.ts [hunk 2]
```diff
@@ -72,17 +71,11 @@ const loginController = (req: TokenRequest, res: Response, next: NextFunction) =
       message: 'Logged In Successfully',
       data: [userInfo],
     }))
-    .catch((err: CustomError | joiInterface) => {
-      if ('isJoi' in err) {
-        next(new CustomError(406, err.details[0].message));
-      } else {
-        next(new CustomError(500, 'server error'));
-      }
-    });
```
Comment (KhaledToha) [inline src/controllers/auth.ts:81]: same as signupController

**Review by KhaledToha — CHANGES_REQUESTED** [review]

File: src/controllers/auth.ts (hunk 2 above)
Comment (Amal-Mousa) [reply src/controllers/auth.ts:81]: ok ع راسي

**Review by KhaledToha — APPROVED** [review]

**Review by nada-abuzaid — APPROVED** [review]

**Pull request merged**